packages/server/lib/util/settings.js

```diff
@@ -144,29 +203,53 @@ module.exports = {
 
     const file = this.pathToConfigFile(projectRoot, options)
 
-    return fs.readJsonAsync(file)
-    .catch({ code: 'ENOENT' }, () => {
-      return this._write(file, {})
-    }).then((json = {}) => {
-      if (this.isComponentTesting(options) && 'component' in json) {
-        json = { ...json, ...json.component }
+    return requireAsync(file,
```

Comment:
What is the plan for the existing plugins code that essentially does the same thing as `requireAsync`? `lib/util/require_async` and `lib/plugins` cover a lot of the same ground. Is `lib/plugins` going to be deleted with all of its functionality eventually migrated to `lib/util/require_async`?

I thought `cypress.config.js` was going to replace the existing plugins code, but this seems to be adding on a separate plugins-like child_process manager. What's the plan?

Reply:
In this PR, I do not implement the plugins API to keep it readable

2 steps remaining after this PR:
- When I will implement the new `setupNodeEvents` API I will keep using the current plugins API. This will avoid having to change may unit test/integration tests.
- When we drop `cypress.json`, I expect to move all the plugin evaluation code to use `require_async`.

Reply:
Okay I see. Makes sense. I think `require_async` is a bit of a misnomer, since we use `child_process` (and presumably plan to continue doing so), but it's not a big deal.